msfvenom

```diff
@@ -69,81 +69,69 @@ def parse_args(args)
+  opt.on('-f', '--format          <format>', String, "Output format (use --list formats to list)") do |f|
     opts[:format] = f
   end
 
-  opt.on('--help-formats', String, "List available formats") do
-    init_framework(:module_types => [])
-    msg = "Executable formats\n" +
-      "\t" + ::Msf::Util::EXE.to_executable_fmt_formats.join(", ") + "\n" +
-      "Transform formats\n" +
-      "\t" + ::Msf::Simple::Buffer.transform_formats.join(", ")
-    raise HelpError, msg
+  opt.on('-e', '--encoder         <encoder>', String, 'The encoder to use (use --list encoders to list)') do |e|
+    opts[:encoder] = e
   end
 
-  opt.on('--encrypt         <value>', String, 'The type of encryption or encoding to apply to the shellcode') do |e|
-    opts[:encryption_format] = e
+  opt.on('--smallest', 'Generate the smallest possible payload using all available encoders') do
+    opts[:smallest] = true
   end
 
-  opt.on('--encrypt-formats', String, 'List Available encryption formats') do
-    init_framework(:module_types => [])
-    msg = "Encryption formats:\n" +
-          "\t" + ::Msf::Simple::Buffer.encryption_formats.join(", ")
-    raise HelpError, msg
+  opt.on('--encrypt         <value>', String, 'The type of encryption or encoding to apply to the shellcode (use --list encrypt to list)') do |e|
+    opts[:encryption_format] = e
   end
 
-  opt.on('--encrypt-key     <value>', String, 'A key to be used for the encryptor') do |e|
+  opt.on('--encrypt-key     <value>', String, 'A key to be used for --encrypt') do |e|
     opts[:encryption_key] = e
   end
 
-  opt.on('--encrypt-iv      <value>', String, 'An initialization vector for the encryption') do |e|
+  opt.on('--encrypt-iv      <value>', String, 'An initialization vector for --encrypt') do |e|
     opts[:encryption_iv] = e
   end
 
-  opt.on('-e', '--encoder         <encoder>', String, 'The encoder to use') do |e|
-    opts[:encoder] = e
-  end
-
-  opt.on('-a', '--arch            <arch>', String, 'The architecture to use') do |a|
+  opt.on('-a', '--arch            <arch>', String, 'The architecture to use for --payload and --encoders') do |a|
     opts[:arch] = a
   end
 
-  opt.on('--platform        <platform>', String, 'The platform of the payload') do |l|
+  opt.on('--platform        <platform>', String, 'The platform for --payload (use --list platforms to list)') do |l|
     opts[:platform] = l
   end
 
-  opt.on('--help-platforms', String, 'List available platforms') do
-    init_framework(:module_types => [])
-    supported_platforms = []
-    Msf::Module::Platform.subclasses.each {|c| supported_platforms << "#{c.realname.downcase}"}
-    msg = "Platforms\n" +
-      "\t" + supported_platforms.sort * ", "
-    raise HelpError, msg
+  opt.on('-o', '--out             <path>', 'Save the payload to a file') do |x|
+    opts[:out] = x
+  end
+
+  opt.on('-b', '--bad-chars       <list>', String, 'Characters to avoid example: \'\x00\xff\'') do |b|
```

Comment:
Being pedantic.

This
`"Characters to avoid example: '\\x00\\xff'"`
is easier to read than
`'Characters to avoid example: \'\x00\xff\''`

Reply:
I personally don't agree with it being easier to read.

Reply:
I think they both look like crap.